### PR TITLE
fix nf_elem_fmpz_sub

### PR DESCRIPTION
**Pull request by fieker**
seems to work better - copied from the si_sub file....

### CODE REVIEW

File: nf_elem/fmpz_sub.c [hunk 1]
```diff
@@ -41,15 +41,19 @@ void nf_elem_fmpz_sub(nf_elem_t a, const fmpz_t c, const nf_elem_t b, const nf_t
...
 	  while (len != 0 && fmpz_is_zero(num + len - 1))
 	     len--;
-	  
-      fmpz_submul(num, den, c);
-	  fmpz_neg(num, num);
+
+
+          fmpz_submul(num, den, c);
+
+          for(i = 0; i <= len; i++)
```
Comment (wbhart) [inline nf_elem/fmpz_sub.c]: Why is is <= instead of < ? I'm probably just missing something obvious.

**Review by wbhart — CHANGES_REQUESTED** [review]

**Comment by fieker** [comment]
On Fri, Aug 23, 2019 at 07:11:39AM -0700, wbhart wrote:
> wbhart requested changes on this pull request.
> 
> 
> 
> >  	  
>  	  nf_elem_set(a, b, nf);
>  	  
>  	  while (len != 0 && fmpz_is_zero(num + len - 1))
>  	     len--;
> -	  
> -      fmpz_submul(num, den, c);
> -	  fmpz_neg(num, num);
> +
> +
> +          fmpz_submul(num, den, c);
> +
> +          for(i = 0; i <= len; i++)
> 
> Why is is <= instead of < ? I'm probably just missing something obvious.
> 
the while loop ends with len being the last non-zero term. Thus it needs
negating.
Other answer: it is copied from the nf_elem_si_sub case

**Comment by wbhart** [comment]
Consider len = 1. Then num[2 - 1] would have been zero, i.e. there would be one nonzero coefficient, namely num[0].

**Comment by fieker** [comment]
I think it is even more hilariously wrong: there is no len for quadratics.
Quadratics are a strange structure....
I'll have a look again.

**Review by wbhart — APPROVED** [review]

**Pull request merged**